File: src/main/ATOMScopeType.java
package main;

public enum ATOMScopeType {

    PARENTHESIS('(',')'),
    ARRAY('[',']'),
    OBJECT('{','}');

    public char opening;
    public char closing;

    ATOMScopeType(char opening, char closing) {
        this.opening = opening;
        this.closing = closing;
    }

}

File: web_interpreter/atom/ATOMScopeType.js

export default class ATOMScopeType{

  static opening(type) {
    if (type=='PARENTHESIS') {
      return '(';
    }
    if (type=='ARRAY') {
      return '[';
    }
    if (type=='OBJECT') {
      return '{';
    }
    throw 'Unknown type ' + type;
  }

  static closing(type) {
    if (type=='PARENTHESIS') {
      return ')';
    }
    if (type=='ARRAY') {
      return ']';
    }
    if (type=='OBJECT') {
      return '}';
    }
    throw 'Unknown type ' + type;
  }

}